Release/16122022/NSRetail_14122022.sql

```diff
@@ -0,0 +1,1523 @@
+@BREFUNDID INT,        
+@COUNTERID INT,        
+@USERID INT        
+          
+AS            
+BEGIN            
+
+IF EXISTS(SELECT 1 FROM @dtbrd WHERE ISNULL(DealerID,0) = 0 AND (REASONID = 1 OR REASONID = 2))
+BEGIN
+
+SELECT 'Few items does not have supplier information!'
+return;
+
+END
+
+DECLARE @BRANCHID INT    
+SELECT @BRANCHID = BRANCHID FROM BRANCHCOUNTER WHERE COUNTERID = @COUNTERID    
+      
+--Change BRefund status to accepted      
+UPDATE POS_BREFUND SET         
+IsAccepted = 1,        
+UPDATEDBY = @USERID,        
+UPDATEDDATE = GETDATE()        
+WHERE COUNTERID = @COUNTERID        
+AND BREFUNDID = @BREFUNDID        
+      
+--updating accepted quantity and weight in kgs      
+UPDATE BRD          
+SET           
+BRD.ACCEPTEDQUANTITY = UBRD.ACCEPTEDQUANTITY,          
+BRD.ACCEPTEDWEIGHTKGS = UBRD.ACCEPTEDWEIGHTINKGS,
+BRD.SUPPLIERID = UBRD.DEALERID,
+BRD.ITEMCOSTPRICEID = UBRD.ITEMCOSTPRICEID
+FROM POS_BREFUNDDETAIL BRD          
+INNER JOIN @dtbrd UBRD ON BRD.BREFUNDID = UBRD.BREFUNDID          
+AND BRD.BREFUNDDETAILID = UBRD.BREFUNDDETAILID          
+AND BRD.COUNTERID = UBRD.COUNTERID          
+          
+--Dedutcing quantity from branch quantity      
+    
+	UPDATE SS    
+	SET 
+		SS.QUANTITY = SS.QUANTITY - UBRD.ACCEPTEDQUANTITY,    
+		SS.WEIGHTINKGS = SS.WEIGHTINKGS - UBRD.ACCEPTEDWEIGHTINKGS,
+		SS.TTYPE =  'BR'
+	FROM 
+		STOCKSUMMARY SS 
+		INNER JOIN 
+		(
+			SELECT CASE WHEN IP.BRANCHID IS NULL THEN BRD.ITEMPRICEID ELSE IP.PARENTITEMPRICEID END AS ITEMPRICEID
+			, BRD.ACCEPTEDQUANTITY
+			, BRD.ACCEPTEDWEIGHTINKGS
+			FROM ITEMPRICE IP 
+				INNER JOIN @dtbrd BRD ON IP.ITEMPRICEID = BRD.ITEMPRICEID
+		) UBRD ON SS.ITEMPRICEID = UBRD.ITEMPRICEID
+	WHERE SS.BRANCHID = @BRANCHID
+      
+--adding quantity to warehouse      
+	UPDATE SS        
+	SET 
+		SS.QUANTITY = ISNULL(SS.QUANTITY, 0) + SDD.QUANTITY,        
+		SS.WEIGHTINKGS = ISNULL(SS.WEIGHTINKGS, 0) + SDD.WEIGHTINKGS        ,
+		SS.TTYPE =  'BR'
+	FROM 
+		STOCKSUMMARY SS 
+		INNER JOIN          
+			(
+				SELECT           
+					CASE WHEN IP.BRANCHID IS NULL THEN IP.ITEMPRICEID ELSE IP.PARENTITEMPRICEID END AS ITEMPRICEID
+					, ACCEPTEDQUANTITY AS QUANTITY
+					, ACCEPTEDWEIGHTINKGS AS WEIGHTINKGS
+					, BRD.REASONID
+				FROM @dtbrd BRD     
+					INNER JOIN ITEMPRICE IP ON BRD.ITEMPRICEID = IP.ITEMPRICEID          
+					INNER JOIN ITEMCODE IC ON IP.ITEMCODEID = IC.ITEMCODEID          
+					INNER JOIN ITEM I ON IC.ITEMID = I.ITEMID     
+				WHERE (I.PARENTITEMID IS NULL OR ISNULL(I.ISOPENITEM, 0) = 1)     
+			) SDD ON SS.ITEMPRICEID = SDD.ITEMPRICEID
+	WHERE SS.BRANCHID = 45 AND (SDD.REASONID = 3 OR SDD.REASONID = 4)
+
+-- INSERTING SUPPLIER RETURNS RECORDS
+
+	INSERT INTO SUPPLIERRETURNS(SUPPLIERID,CREATEDBY,CREATEDDATE,STATUS)
+	SELECT DISTINCT DEALERID,@USERID,GETDATE(),0 FROM @dtbrd BRD
+	WHERE NOT EXISTS(SELECT 1 FROM SUPPLIERRETURNS SR
+	WHERE SR.SUPPLIERID = BRD.DEALERID AND ISNULL(SR.STATUS,0) = 0)
```

Comment:
add delete date

Reply:
There is not Deleted Date for supplier returns